source/common/stats/stats_impl.h

```diff
@@ -333,27 +339,62 @@ class GaugeImpl : public Gauge, public MetricImpl {
   // Stats::Gauge
   virtual void add(uint64_t amount) override {
     data_.value_ += amount;
-    data_.flags_ |= RawStatData::Flags::Used;
+    data_.flags_ |= Flags::Used;
   }
   virtual void dec() override { sub(1); }
   virtual void inc() override { add(1); }
   virtual void set(uint64_t value) override {
     data_.value_ = value;
-    data_.flags_ |= RawStatData::Flags::Used;
+    data_.flags_ |= Flags::Used;
   }
   virtual void sub(uint64_t amount) override {
     ASSERT(data_.value_ >= amount);
     ASSERT(used());
     data_.value_ -= amount;
   }
   virtual uint64_t value() const override { return data_.value_; }
-  bool used() const override { return data_.flags_ & RawStatData::Flags::Used; }
+  bool used() const override { return data_.flags_ & Flags::Used; }
 
 private:
   RawStatData& data_;
   RawStatDataAllocator& alloc_;
 };
 
+/**
+ * Implementation of HistogramStatistics for circllhist.
+ */
+class HistogramStatisticsImpl : public HistogramStatistics {
+public:
+  HistogramStatisticsImpl() : computed_quantiles_(supported_quantiles_.size(), 0.0) {}
+
+  HistogramStatisticsImpl(histogram_t* histogram_ptr)
+      : computed_quantiles_(supported_quantiles_.size(), 0.0) {
+    hist_approx_quantile(histogram_ptr, supported_quantiles_.data(), supported_quantiles_.size(),
+                         computed_quantiles_.data());
+  }
+
+  std::string summary() const override {
+    std::stringstream summary_stream;
+    size_t index = 0;
+    for (double quantile : supported_quantiles_) {
+      summary_stream << "P" << quantile * 100 << ": " << computed_quantiles_[index];
+      if (index < supported_quantiles_.size() - 1) {
+        summary_stream << ", ";
```

Comment:
much nicer.  WDYT of using absl::StrJoin and fmt::format as a slight simplification.

```
  std::vector<std::string> summary;
  for (size_t i= 0; i< supported_quantiles_.size(); ++i) {
    summary.push_back(fmt::format("P{}: {}", 100*supported_quantiles_[i], computed_quantiles_[i]);
  }
  return absl::StrJoin(summary, ", ");
```
5 lines instead of 10, fwiw.